syz-manager/manager.go

```diff
@@ -1195,3 +1210,192 @@ func publicWebAddr(addr string) string {
 	}
 	return "http://" + addr
 }
+
+func (mgr *Manager) initWeightedPCs() {
```

Comment:
syz-manager.go appeared as a monolith, then it grew to a frightening size and then I spent some time splitting it into pieces. Please don't throw everything into syz-manager.go. Create a helper type in a separate file that will handle of the logic related to the filter.
Also tests are highly encourages, or an explanation why it's not testable at all in the current testing infrastructure.